src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,292 @@
+ *
+ * @see https://laravel.com/docs/5.7/dusk#available-assertions
+ */
+trait WebTestAssertions
+{
+    /** @var Client|null */
+    protected static $client;
+
+    public static function assertResponseIsSuccessful(): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isSuccessful(),
+            "Response was expected to be successful, but actual HTTP code is {$response->getStatusCode()}."
+        );
+    }
+
+    public static function assertHttpCodeEquals(int $expectedCode): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertSame(
+            $expectedCode,
+            $code = $response->getStatusCode(),
+            "Response code [$expectedCode] does not match actual HTTP code [$code]."
+        );
+    }
+
+    public static function assertResponseRedirects(string $expectedLocation = null): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isRedirect(),
+            "Actual response code [{$response->getStatusCode()}] does not match any valid HTTP redirection code."
+        );
+
+        if ($expectedLocation) {
+            Assert::assertSame(
+                $expectedLocation,
+                $location = $response->headers->get('Location'),
+                "Location [$expectedLocation] does not match actual redirection URL [$location]."
+            );
+        }
+    }
+
+    public static function assertPageTitleEquals(string $expectedTitle): void
+    {
+        self::checkCrawlerAvailable();
+
+        $titleNode = static::$client->getCrawler()->filter('title');
+
+        Assert::assertGreaterThan(0, $titleNode->count(), 'There is no <title> tag in the current page to look for.');
+
+        Assert::assertEquals(
+            $expectedTitle,
+            \trim($title = $titleNode->text()),
+            "Expected title [$expectedTitle] does not equal actual title [$title]."
+        );
+    }
+
+    public static function assertPageTitleContains(string $expectedTitle): void
+    {
+        self::checkCrawlerAvailable();
+
+        $titleNode = static::$client->getCrawler()->filter('title');
+
+        Assert::assertGreaterThan(0, $titleNode->count(), 'There is no <title> tag in the current page to look for.');
+
+        Assert::assertContains(
+            $expectedTitle,
+            \trim($title = $titleNode->text()),
+            "Expected title [$expectedTitle] does not equal actual title [$title]."
+        );
+    }
+
+    public static function assertClientHasCookie(string $name, string $path = '/', string $domain = null): void
+    {
+        self::checkClientAvailable();
+
+        $cookie = static::$client->getCookieJar()->get($name, $path, $domain);
+
+        Assert::assertNotNull(
```

Comment:
Maybe return which cookies did exist? In case someone made a type.

Reply:
Not sure, if the developer made a typo they will see it in the error message, since it's added in the assertion exception message